standards/SPIFFE-ID.md

```diff
@@ -185,3 +195,29 @@ For example, imagine that the operator of trust domain A and trust domain B have
 
+### 4.2. ID Equivalency
+The comparison of SPIFFE IDs is a security critical operation. In allowing for internationalization, as well as compatibility with the DNS system, the URI standard which SPIFFE IDs rely upon has made a handful of decisions which complicate the equivalency process. Specifically, case-insensitivity, ambiguous percent-encoding rules and support for UTF-8, and the implementation of many conditionally-special characters can make the topic a confusing one.
+
+This specification has specifically forbidden the use of characters and encoding schemes that complicate comparison (please see sections [2.1](#21-trust-domain) and [2.2](#22-path) for more information). Thus, any legal SPIFFE ID can be considered safe. It will not be misinterpreted or transformed by traditional URI libraries, and it can be safely compared with other legal IDs on a byte-for-byte basis.
+
+It is still possible for a legal URI (but illegal SPIFFE ID) to be legally processed according to URI normalization rules _into_ a legal SPIFFE ID. Thus, under certain circumstances, it is possible for an illegal SPIFFE ID to be passed through a URI parser and produce a legal SPIFFE ID. These two SPIFFE IDs **will not match** on a byte-for-byte basis, however there exists room for confusion.
+
+This is an important detail to keep in mind, particularly when accepting untrusted input that is not signed over by a trust domain authority (e.g. user input from a web app). Untrusted input should always be validated against the SPIFFE ID ruleset prior to accepting and parsing with a traditional URI parser. Please see [Appendix A](#5-appendix-a) for a brief example.
+
+## 5. Appendix A. Lightweight SPIFFE ID validation
+SPIFFE IDs have a strict character set that is designed to be as consistent and as easy to validate as possible. Here is a logic example demonstrating a simple SPIFFE ID validation mechanism:
+1. Total number of characters is less than 2048
+1. Starts with "spiffe://"
+1. Strip prefix "spiffe://"
+1. The (new) first character must not be a `/`
+1. From the beginning, strip one character at a time:
+    1. Ensure it is either a lowercase alpha-numeric, a `.`, a `-`, or a `/`
+    1. When `/` is detected, stop
+    1. The number of stripped characters should not exceed 255, not including `/`
+1. What remains is the path - scan it for the following conditions in sequential order:
+    1. The string does not start or end with `/`
+    1. Includes only uppercase or lowercase alpha-numeric characters, `.`s, `-`s, `_`s, and `/`s
+    1. The character sequences `//`, `/./`, and `/../` do not appear anywhere in the string
+
+If the above checks are successful, the SPIFFE ID is valid.
```

Comment:
I think @spikecurtis's python script can be linked to from here. Maybe we can add it to this PR and check it into a new `/standards/examples` directory?

Reply:
Alternatively, we could link to it directly from section 4.2 and omit this appendix item

Reply:
> Maybe we can add it to this PR and check it into a new /standards/examples directory?

Makes sense. 

I was also thinking, maybe it'd be useful to have a `validate()` function in `go-spiffe` — like a static function that takes a String and return true/false based on whether the ID is standards-conformant. It can even have a "strict" flag that treats "SHOULD"s as "MUST" (hence ensuring the length does not exceed 2048 chars—which is, normally, a recommendation, not a requirement)